src/coreclr/jit/emit.cpp

```diff
@@ -4102,6 +4199,382 @@ void emitter::emitDispCommentForHandle(size_t handle, GenTreeFlags flag)
+            previousJumpIgNo  = jmp->idjIG->igNum;
+            previousJumpInsNo = jmp->idDebugOnlyInfo()->idNum;
+#endif
+
+            // target group is not bound yet so use the cookie to fetch it
+            insGroup* targetGroup = (insGroup*)emitCodeGetCookie(jmp->idAddr()->iiaBBlabel);
+
+            if (targetGroup != nullptr && (jmp->idjIG->igFlags & IGF_HAS_ALIGN) == 0 &&
+                jmp->idjIG->igNext == targetGroup)
+            {
+                insGroup* group            = jmp->idjIG;
+                unsigned  instructionCount = group->igInsCnt;
+
+                if (instructionCount > 0)
+                {
+                    BYTE*      dataPtr               = group->igData;
+                    instrDesc* instructionDescriptor = nullptr;
+
+                    while (instructionCount > 0)
+                    {
+                        instructionDescriptor = (instrDesc*)dataPtr;
+                        dataPtr += emitSizeOfInsDsc(instructionDescriptor);
+                        instructionCount -= 1;
+                    };
+
+                    assert(instructionDescriptor != nullptr);
+                    // instructionDescriptor is now the last instruction in the group
+
+                    if (instructionDescriptor->idIns() == jmp->idIns() && jmp == instructionDescriptor)
+                    {
+                        CLANG_FORMAT_COMMENT_ANCHOR
+// the last instruction in the group is the jmp we're looking for
+// and it jumps to the next instruction group so we don't really need it
+
+#ifdef DEBUG
+                        if (EMITVERBOSE)
+                        {
+                            printf("IN%04x from the last instruction in IG%02u and jumps to the next IG IG%02u, "
+                                   "dropping\n",
+                                   jmp->idDebugOnlyInfo()->idNum, group->igNum, targetGroup->igNum);
+                        }
+#endif
+
+                        if (previousTruncatedGroup)
+                        {
+                            for (insGroup* offsetGroup = previousTruncatedGroup->igNext; offsetGroup != group->igNext;
+                                 offsetGroup           = offsetGroup->igNext)
+                            {
+#ifdef DEBUG
+                                if (EMITVERBOSE)
+                                {
+                                    printf("Adjusted offset of IG%02u from %04X to %04X\n", offsetGroup->igNum,
+                                           offsetGroup->igOffs, (offsetGroup->igOffs - removedSize));
+                                }
+#endif
+                                offsetGroup->igOffs -= removedSize;
+                                assert(IsCodeAligned(offsetGroup->igOffs));
+                            }
+                        }
+
+                        previousTruncatedGroup = group;
+
+                        UNATIVE_OFFSET sizeRemoved = instructionDescriptor->idCodeSize();
```

Comment:
I think you forgot to do `removedSize += sizeRemoved`, although I would definitely rename those variables to something meaningful like `totalSizeRemoved` and `currentJmpSizeRemoved` or something like that.

Reply:
Yes, this was confused by trying to convert it to linear form. it's cleaned up.